ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/workspace/event/MachineStatusChangedEvent.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.ide.api.workspace.event;
+
+import com.google.gwt.event.shared.EventHandler;
+import com.google.gwt.event.shared.GwtEvent;
+import org.eclipse.che.api.core.model.workspace.runtime.MachineStatus;
+
+/** Fired when state of machine has been changed. */
+public class MachineStatusChangedEvent extends GwtEvent<MachineStatusChangedEvent.Handler> {
```

Comment:
I know it's better to use events you proposed.
As we have discussed there is a [bug](https://github.com/eclipse/che/issues/7900) that after receiving `machine running` event IDE immediately reads the workspace state but this machine is absent in runtime and `MachineStartingEvent` is not sent.

Reply:
I don't understand you motivation create new one Event if we already have them

Reply:
I see. Thanks for the explanation.
If it's a bug of the server-side that should be fixed, I think no need to introduce `MachineStatusChangedEvent` as part of IDE API but leave it in `che-core-ide-app` module where it's actually used to prevent widespread use of this "temporary" event by plugins.

Reply:
MachineStartingEvent is never sent.

Reply:
@azatsarynnyy Great idea. I'm moving the event to `chde-core-ide-app` and making a [note](https://github.com/eclipse/che/issues/8028) I have to return back to this event after fixing a [bug](https://github.com/eclipse/che/issues/7900)

Reply:
Done